crud/common/sharding.lua

```diff
@@ -20,14 +27,29 @@ function sharding.tuple_get_bucket_id(tuple, space, specified_bucket_id)
         return specified_bucket_id
     end
 
-    local key = utils.extract_key(tuple, space.index[0].parts)
+    local primary_index = space.index[0]
+    local key
+    local replicasets = vshard.router.routeall()
+    local replicaset = select(2, next(replicasets))
+    local sharding_key = sharding.get_ddl_sharding_key(space.name, replicaset.master.conn)
+    if sharding_key ~= nil then
+        local sharding_keys_fieldnos = utils.get_keys_fieldnos(space:format(), sharding_key)
+        if sharding_keys_fieldnos ~= nil then
+            key = utils.extract_sharding_key(tuple, sharding_keys_fieldnos)
+        end
+    end
+
+    if key == nil then
+        key = utils.extract_key(tuple, primary_index.parts)
+    end
```

Comment:
Finally we discussed an approach and I expect you will return with construction like

```lua
local sharding_key_parts
 
if <we can use ddl> then
    -- create something similar to index parts
    -- but based on ddl sharding key
else
    sharding_key_parts = primary_index.parts    
end

key = utils.extract_key(tuple, primary_index.parts)
```

Reply:
Did it, I'll push update soon.